### PR TITLE
Add dockerhub proxy to CI

### PR DESCRIPTION
**Pull request by alesapin**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

cc @qoega

### CODE REVIEW

**Comment by alesapin** [comment]
Proxy works, checked in logs.

**Comment by alesapin** [comment]
```172.31.33.105 - - [17/Dec/2021:12:41:43 +0000] "GET /v2/yandex/clickhouse-server/blobs/sha256:a70d879fa5984474288d52009479054b8bb2993de2a1859f43b5480600cecb24 HTTP/1.1" 200 28569016 "" "docker/20.10.12 go/go1.16.12 git-commit/459d0df kernel/5.11.0-1022-aws os/linux arch/amd64 UpstreamClient(docker-compose/1.28.2 docker-py/5.0.3 Linux/5.11.0-1022-aws)"
time="2021-12-17T12:41:44.180998999Z" level=info msg="response completed" go.version=go1.11.2 http.request.host="dockerhub-proxy.dockerhub-proxy-zone:5000" http.request.id=e4e49c23-1732-4015-a530-5a47fc1a950a http.request.method=GET http.request.remoteaddr="172.31.33.105:48752" http.request.uri="/v2/yandex/clickhouse-server/blobs/sha256:ed70af9ad80291d53ed229a2982edd5b4e34027ff08e6fbbd46fece31d9e8c3d" http.request.useragent="docker/20.10.12 go/go1.16.12 git-commit/459d0df kernel/5.11.0-1022-aws os/linux arch/amd64 UpstreamClient(docker-compose/1.28.2 docker-py/5.0.3 Linux/5.11.0-1022-aws)" http.response.contenttype="application/octet-stream" http.response.duration=651.468945ms http.response.status=200 http.response.written=145516599 
172.31.33.105 - - [17/Dec/2021:12:41:43 +0000] "GET /v2/yandex/clickhouse-server/blobs/sha256:ed70af9ad80291d53ed229a2982edd5b4e34027ff08e6fbbd46fece31d9e8c3d HTTP/1.1" 200 145516599 "" "docker/20.10.12 go/go1.16.12 git-commit/459d0df kernel/5.11.0-1022-aws os/linux arch/amd64 UpstreamClient(docker-compose/1.28.2 docker-py/5.0.3 Linux/5.11.0-1022-aws)"
```

**Comment by alesapin** [comment]
```E               Pulling node3 (yandex/clickhouse-server:20.3.16)...
E               Get "https://registry-1.docker.io/v2/": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
```

**Review by Felixoid — APPROVED** [review]

File: docker/test/integration/runner/dockerd-entrypoint.sh [hunk 1]
```diff
@@ -8,8 +8,8 @@ echo '{
     "ip-forward": true,
     "log-level": "debug",
     "storage-driver": "overlay2",
-    "insecure-registries" : ["dockerhub-proxy.sas.yp-c.yandex.net:5000"],
-    "registry-mirrors" : ["http://dockerhub-proxy.sas.yp-c.yandex.net:5000"]
+    "insecure-registries" : ["dockerhub-proxy.dockerhub-proxy-zone:5000"],
```
Comment (Alex-Cheng) [inline docker/test/integration/runner/dockerd-entrypoint.sh:11]: `dockerhub-proxy.dockerhub-proxy-zone`  seems not to be a valid URL, shall we set it in my local hosts file?

Comment (Felixoid) [reply docker/test/integration/runner/dockerd-entrypoint.sh:11]: If the host is not found it's just not used, AFAIR

Comment (Alex-Cheng) [reply docker/test/integration/runner/dockerd-entrypoint.sh:11]: so does it mean the host is never used? but I see the log about trying to connect to the host but failed.

Comment (Felixoid) [reply docker/test/integration/runner/dockerd-entrypoint.sh:11]: does it say "served from upstream"? Or something like that?

Comment (Alex-Cheng) [reply docker/test/integration/runner/dockerd-entrypoint.sh:11]: I got the following logs:
image

Comment (Felixoid) [reply docker/test/integration/runner/dockerd-entrypoint.sh:11]: Lines 511-512

**Pull request merged**